### PR TITLE
Adds awareness traits

### PR DESCRIPTION
**Pull request by mecha**

### CODE REVIEW

**Review by XedinUnknown — CHANGES_REQUESTED** [review]
What about the [subject][]?

[subject]: https://github.com/Dhii/data-state-base/pull/1/files#diff-1f837cf677eb707ec3559163e3750162R115

**Comment by mecha** [comment]
Coming soon to a cinema near you

File: src/StateAwareAwareTrait.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    protected $stateAware;
+
+    /**
+     * Retrieves the state-aware subject associated with this instance.
+     *
+     * @since [*next-version*]
+     *
+     * @return StateAwareInterface The state-aware subject instance.
+     */
+    protected function _getStateAware()
```
Comment (XedinUnknown) [inline src/StateAwareAwareTrait.php]: Cannot be null?

**Review by XedinUnknown — CHANGES_REQUESTED** [review]

File: src/StateAwareAwareTrait.php (hunk 1 above)
Comment (mecha) [reply src/StateAwareAwareTrait.php]: Sorry can. Doc is wrong.

**Review by XedinUnknown — APPROVED** [review]

**Pull request merged**